src/Shared/Microsoft.DotNet.Kusto/KustoQuery.cs

```diff
@@ -4,34 +4,31 @@
 using System.Collections.Generic;
 using System.Linq;
 
-namespace Microsoft.DotNet.Kusto;
-
-public class KustoQuery
+namespace Microsoft.DotNet.Kusto
 {
-    public KustoQuery()
+    public class KustoQuery
     {
-        Parameters = new List<KustoParameter>();
-    }
-
-    public KustoQuery(string text, IEnumerable<KustoParameter> parameters)
-    {
-        Text = text;
-        Parameters = parameters.ToList();
-    }
+        public KustoQuery(string text = null, IEnumerable<KustoParameter> parameters = null, IEnumerable<KustoQueryOption> options = null)
+        {
+            Text = text;
+            Parameters = parameters?.ToList() ?? new List<KustoParameter>();
+            Options = options?.ToList() ?? new List<KustoQueryOption>();
```

Comment:
I don't see this property defined in this class...  Maybe I'm just reading the diff wrong, but where is "Options" defined?  A random string=>object dictionary is really hard to use... how do I know what values to pass to it?

Where are these values going?  Can defined them more usefully, at least for options we support?  I still feel like the options probably don't need to be passed into the Query (especially as an IEnumerable), but rather to the actual "ExecuteQuery" method (since you are likely to use the same options for multiple requests, there no real reason to have to copy it into all the queries).